express-crud/server.ts

```diff
@@ -0,0 +1,139 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+    select *
+    from "grades";
+    `;
+    const result = await db.query(sql);
+    res.status(200).json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades:gradeId', async (req, res, next) => {
+  try {
+    const { gradeId } = req.params;
+    if (gradeId === undefined) {
+      throw new ClientError(400, 'gradeId is required');
+    }
+    const sql = `
+    select *
+    from "grades"
+    where "gradeId" = $1
+    `;
+    const result = await db.query(sql, [gradeId]);
+    const film = result.rows[0];
+    res.status(200).json(film);
```

Comment:
You need to check for the error _before_ sending the result back. Otherwise, you will try to send a good response, and then a bad one. Express doesn't like that!